src/hooks/useENSAssignRegistration.ts

```diff
@@ -0,0 +1,193 @@
+import { differenceWith, isEqual } from 'lodash';
```

Comment:
love this

Reply:
thoughts on `useENSModifyRegistration` as a name?

Reply:
hmm, maybe a verb doesn't make sense since we aren't performing any actions in this hook...

what about `useENSModifiedRegistration`?

Reply:
sounds 👌